controllers/user.js

```diff
@@ -1,14 +1,54 @@
-const { loginUser, registerUser } = require("../services/user");
+const {
+  loginUser,
+  registerUser,
+  createToken,
+  comparePasswords,
+} = require("../services/user");
+
+const handleError = (error) => {
+  const errors = { email: "", password: "" };
+
+  if (error.code === 11000) {
+    errors.email = "Email already exists";
+    return errors;
+  }
+
+  errors.email = error?.errors?.email?.message;
+  errors.password = error?.errors?.password?.message;
+  return errors;
+};
 
 const login = async (req, res) => {
-  await loginUser();
-  res.status(200).send("logged in");
+  try {
+    const user = await loginUser(req.body);
+    //checking if user exists in DB
+    if (!user) {
+      return res.status(400).json({ message: "No user found with such email" });
+    }
+    //checking if passwords are correct
+    const isPasswordCorrect = await comparePasswords(
+      req.body.password,
+      user.password
+    );
+    if (!isPasswordCorrect) {
+      return res.status(400).json({ message: "Incorrect password" });
```

Comment:
Messages 'Incorrect password'  and 'No user found with such email' are not secured. It's better to use single message for both cases, f.e. 'Wrong user email or password'.
Error 'Incorrect password' says that email is correct by password - not and somebody can try to guess the password

Reply:
And i thought why i see message like 'Wrong user email or password' on web pages, because then i have to guess which one is incorrect, now i know :d i'll change that too